static/handle-approval.ts

```diff
@@ -0,0 +1,116 @@
+const approveButton = document.querySelector(".approve-button") as HTMLButtonElement;
+const revokeButton = document.querySelector(".revoke-button") as HTMLButtonElement;
+
+function isValidAddress(): boolean {
+  // Check if the address is 20 bytes long (40 characters) excluding the '0x' prefix
+  const result = /^0x[a-fA-F0-9]{40}$/.test(addressInput.value);
+
+  console.log("a");
+
+  if (result) {
+    addressInput.style.border = "1px solid #5af55a";
+  } else if (addressInput.value === "") {
+    addressInput.style.border = "1px solid rgba(255, 255, 255, 0.1)";
+  } else {
+    addressInput.style.border = "1px solid red";
+  }
+
+  return result;
+}
+
+function isValidAmount(): boolean {
+  // Check if the amount is a positive number
+  const result = !isNaN(Number(amountInput.value)) && Number(amountInput.value) > 0;
+
+  if (result) {
+    amountInput.style.border = "1px solid #5af55a";
+  } else if (amountInput.value === "") {
+    amountInput.style.border = "1px solid rgba(255, 255, 255, 0.1)";
+  } else {
+    amountInput.style.border = "1px solid red";
+  }
+
+  return result;
+}
+
+function isApprovalValid() {
+  const addressValidity = isValidAddress();
+  const amountValidity = isValidAmount();
+  const isValid = addressValidity && amountValidity;
+
+  approveButton.disabled = !isValid;
+  revokeButton.disabled = !addressValidity;
+
+  if (addressValidity && appState.getIsConnectedState()) {
+    void getCurrentAllowance();
+  }
+}
+
+async function getCurrentAllowance() {
+  const tokenAddress = addressInput.value;
+  const permit2Address = getPermit2Address(appState.getCaipNetworkId() as number);
+  const userAddress = appState.getAddress();
+  const tokenContract = new ethers.Contract(tokenAddress, erc20Abi, provider);
+
+  try {
+    const symbol = await tokenContract.symbol();
+    const decimals = await tokenContract.decimals();
+    const allowance = await tokenContract.allowance(userAddress, permit2Address);
+    const formattedAllowance = ethers.utils.formatUnits(allowance, decimals);
+    currentAllowanceAmount.textContent = formattedAllowance + " " + symbol;
+    return allowance;
+  } catch (error) {
+    console.error("Error fetching allowance:", error);
+    renderErrorInModal(error as Error);
+  }
+}
+
+export function setupApproveButton() {
+  approveButton.addEventListener("click", async () => {
+    const tokenAddress = addressInput.value;
+    const permit2Address = getPermit2Address(appState.getCaipNetworkId() as number);
+    const tokenContract = new ethers.Contract(tokenAddress, erc20Abi, userSigner);
+
+    try {
+      const decimals = await tokenContract.decimals();
+      const amount = ethers.utils.parseUnits(amountInput.value, decimals);
+      const tx = await tokenContract.connect(userSigner).approve(permit2Address, amount);
+      await tx.wait();
+      await getCurrentAllowance();
+    } catch (error) {
```

Comment:
Consider adding a loading spinner or something similar while the transaction is waiting to be mined so the user knows what's going on